### PR TITLE
Add Logging for Test Suite events for JUnit4

### PR DESCRIPTION
**Pull request by Raphael-N**

### CODE REVIEW

File: jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/TestEventInterceptorUtility.java [hunk 1]
```diff
@@ -117,35 +107,46 @@ public static void incrementRunCount() {
...
+        if (currentTestCase.equals(testName)) {
             return;
         }
-        currentTestCase = getTestCaseName(description);
-        currentTestSuite = getTestSuiteName(description);
+        currentTestCase = testName;
+
+        final String suiteName = getTestSuiteName(description);
+        if (!currentTestSuite.equals(suiteName)) {
+            if (!currentTestSuite.equals("")) {
```
Comment (delsner) [inline jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/TestEventInterceptorUtility.java:118]: Is there an .empty() for strings in java?

File: jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestFailedInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import net.bytebuddy.asm.Advice;
+
+public class JUnit4TestFailedInterceptor {
+
+    @Advice.OnMethodEnter
+    public static void enter() {
+        try {
+            TestEventInterceptorUtility.incrementFailureCount();
+        } catch (final Exception e) {
+            System.err.println("Exception in increaseIgnoreCount: " + e.getMessage());
```
Comment (delsner) [inline jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestFailedInterceptor.java:13]: Ignore vs failure

File: jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestIgnoredInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import net.bytebuddy.asm.Advice;
+
+public class JUnit4TestIgnoredInterceptor {
+
+    @Advice.OnMethodEnter
+    public static void enter() {
+        try {
+            TestEventInterceptorUtility.incrementIgnoreCount();
+        } catch (final Exception e) {
+            System.err.println("Exception in increaseFailureCount: " + e.getMessage());
```
Comment (delsner) [inline jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestIgnoredInterceptor.java:13]: Ignore vs failure

File: jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestRunFinishedInterceptor.java [hunk 4]
```diff
@@ -2,15 +2,14 @@
...
+public class JUnit4TestRunFinishedInterceptor {
     @Advice.OnMethodEnter
-    public static void enter(@Advice.Argument(0) final Result result) {
+    public static void enter() {
         try {
-            TestEventInterceptorUtility.testRunFinished(result.getRunCount(), result.getFailureCount(), result.getIgnoreCount());
+            TestEventInterceptorUtility.testSuiteFinished();
         } catch (final Exception e) {
-            System.err.println("Exception in test run finished is: " + result);
+            System.err.println("Exception in test run finished is: " + e.getMessage());
```
Comment (delsner) [inline jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/interceptors/JUnit4TestRunFinishedInterceptor.java:12]: Maybe adjust log message here as well to test suite end

**Comment by delsner** [comment]
Looks good. Have you checked if this fixes #22 ? Did the junit5 and junit4 project still work in forking mode with `test.instr=false` ?

File: jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/TestEventInterceptorUtility.java (hunk 1 above)
Comment (Raphael-N) [reply jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/testevent/TestEventInterceptorUtility.java:118]: .isEmpty() exists it seems. Replace all occurrences.

**Comment by Raphael-N** [comment]
For JUnit4 the test log is empty with test.instr=false but it is on main as well at the moment. Everything else is still working.

**Comment by delsner** [comment]
LGTM. Is there anything missing still or why is this WIP?

**Comment by Raphael-N** [comment]
I just forgot to remove WIP 😄

**Pull request merged**